src/components/VRating/VRating.ts

```diff
@@ -0,0 +1,202 @@
+// Styles
+import '../../stylus/components/_rating.styl'
+
+// Components
+import VIcon from '../VIcon'
+
+// Mixins
+import Colorable from '../../mixins/colorable'
+import Sizeable from '../../mixins/sizeable'
+import Rippleable from '../../mixins/rippleable'
+import Themeable from '../../mixins/themeable'
+
+// Utilities
+import { createRange } from '../../util/helpers'
+import mixins from '../../util/mixins'
+
+// Types
+import { VNode, VNodeDirective, VNodeChildrenArrayContents } from 'vue'
+
+type ItemSlotProps = {
+  index: number
+  value: number
+  isFilled: boolean
+  isHalfFilled: boolean
+  isHovered: boolean
+  isHalfHovered: boolean
+  click: Function
+}
+
+/* @vue/component */
+export default mixins(
+  Colorable,
+  Rippleable,
+  Sizeable,
+  Themeable
+).extend({
+  name: 'v-rating',
+
+  props: {
+    backgroundColor: {
+      type: String,
+      default: 'secondary'
+    },
+    color: {
+      type: String,
+      default: 'primary'
+    },
+    emptyIcon: {
+      type: String,
+      default: '$vuetify.icons.ratingEmpty'
+    },
+    fullIcon: {
+      type: String,
+      default: '$vuetify.icons.ratingFull'
+    },
+    halfIcon: {
+      type: String,
+      default: '$vuetify.icons.ratingHalf'
+    },
+    halfIncrements: Boolean,
+    length: {
+      type: Number,
+      default: 5
+    },
+    readonly: Boolean,
+    showHover: Boolean,
+    value: {
+      type: Number,
+      required: true
+    }
+  },
+
+  data: () => ({
+    hoverIndex: -1
+  }),
+
+  computed: {
+    directives (): VNodeDirective[] {
+      if (this.readonly || !this.ripple) return []
+
+      return [{
+        name: 'ripple',
+        value: { circle: true }
+      } as VNodeDirective]
+    },
+    iconProps (): object {
+      const {
+        dark,
+        medium,
+        large,
+        light,
+        small,
+        xLarge
+      } = this.$props
```

Comment:
missing `default` prop

Reply:
There is no default prop, the icon will have its original size (same functionality as the `v-icon` component) of 24px.

Reply:
oops, i thought i've seen it in v-icon, but apparently not